code/modules/vehicle/tg_vehicles/tg_vehicles.dm

```diff
@@ -0,0 +1,242 @@
+/obj/tgvehicle
+	name = "generic vehicle"
+	desc = "Yell at coderbus."
+	icon = 'icons/obj/tgvehicles.dmi'
+	icon_state = "error"
+	max_integrity = 300
+	armor = list(MELEE = 30, BULLET = 30, LASER = 30, ENERGY = 0, BOMB = 30, RAD = 0, FIRE = 60, ACID = 60)
+	density = TRUE
+	anchored = FALSE
+	blocks_emissive = EMISSIVE_BLOCK_GENERIC
+	buckle_lying = FALSE
+	can_buckle = TRUE
+	buckle_lying = 0
+	pass_flags = PASSTABLE
+	COOLDOWN_DECLARE(message_cooldown)
+	COOLDOWN_DECLARE(cooldown_vehicle_move)
+	var/list/mob/occupants //mob = bitflags of their control level.
+	///Maximum amount of passengers plus drivers
+	var/max_occupants = 1
+	////Maximum amount of drivers
+	var/max_drivers = 1
+	/**
+	  * If the driver needs a certain item in hand (or inserted, for vehicles) to drive this. For vehicles, this must be duplicated on their riding component subtype
+	  * [/datum/component/riding/var/keytype] variable because only a few specific checks are handled here with this var, and the majority of it is on the riding component
+	  * Eventually the remaining checks should be moved to the component and this var removed.
+	  */
+	var/key_type
+	///The inserted key, needed on some vehicles to start the engine
+	var/obj/item/key/inserted_key
+	/// Whether the vehicle is currently able to move
+	var/can_move = TRUE
+	var/list/autogrant_actions_passenger //plain list of typepaths
+	var/list/autogrant_actions_controller //assoc list "[bitflag]" = list(typepaths)
+	var/list/mob/occupant_actions //assoc list mob = list(type = action datum assigned to mob)
+	///This vehicle will follow us when we move (like atrailer duh)
+	var/obj/tgvehicle/trailer
+	var/are_legs_exposed = FALSE
+
+/obj/tgvehicle/Initialize(mapload)
+	. = ..()
+	occupants = list()
+	autogrant_actions_passenger = list()
+	autogrant_actions_controller = list()
+	occupant_actions = list()
+	generate_actions()
+
+/obj/tgvehicle/Destroy(force)
+	QDEL_NULL(trailer)
+	inserted_key = null
+	return ..()
+
+/obj/tgvehicle/Exited(atom/movable/gone, direction)
+	if(gone == inserted_key)
+		inserted_key = null
+	return ..()
+
+/obj/tgvehicle/examine(mob/user)
+	. = ..()
+	. += generate_integrity_message()
+
+/// Returns a readable string of the vehicle's health for examining. Overridden by subtypes who want to be more verbose with their health messages.
+/obj/tgvehicle/proc/generate_integrity_message()
+	var/examine_text = ""
+	var/integrity = obj_integrity/max_integrity * 100
+	switch(integrity)
+		if(50 to 99)
```

Comment:
```suggestion
		if(50 to INFINITY)
```

Reply:
this will make it look damaged even if it is not damaged

Reply:
oh you're so right